mybatis-plus/src/test/java/com/baeldung/mybatisplus/MyBatisPlusIntegrationTest.java

```diff
@@ -0,0 +1,206 @@
+        clientQueryWrapper.allEq(Map.of("email", "x@e.com"));
+
+        assertTrue(clientService.list(clientQueryWrapper)
+            .size() > 5);
+    }
+
+    @Test
+    @Order(7)
+    public void whenRemoveByIdIsInvoked_ThenClientIsDeleted() {
+        clientService.removeById(1);
+
+        assertNull(clientService.getById(1));
+    }
+
+    @Test
+    @Order(7)
+    public void givenColumnMap_whenRemoveByMapIsInvoked_ThenClientsAreDeleted() {
+        Map<String, Object> columnMap = new HashMap<>();
+        columnMap.put("email", "x@e.com");
+
+        clientService.removeByMap(columnMap);
+
+        assertEquals(0, clientService.list()
+            .size());
+    }
+
+    @Test
+    @Order(8)
+    public void whenSaveBatchIsInvoked_ThenClientsAreInsertedInBatch() {
+        Client client2 = new Client();
+        client2.setFirstName("Harry");
+
+        Client client3 = new Client();
+        client3.setFirstName("Ron");
+
+        Client client4 = new Client();
+        client4.setFirstName("Hermione");
+
+        clientService.saveBatch(Arrays.asList(client2, client3, client4));
+
+        assertNotNull(client2.getId());
+        assertNotNull(client3.getId());
+        assertNotNull(client4.getId());
+    }
+
+    @Test
+    @Order(9)
+    public void givenExistingClients_whenPageIsInvoked_ThenResultsArePaginated() {
+        Client client2 = new Client();
+        client2.setFirstName("John");
+
+        Client client3 = new Client();
+        client3.setFirstName("Smith");
+
+        Client client4 = new Client();
+        client4.setFirstName("Charlie");
+
+        clientService.saveBatch(Arrays.asList(client2, client3, client4));
+
+        Page<Client> page = Page.of(2, 3);
+
+        assertEquals(3, clientService.page(page, null)
+            .getRecords()
+            .size());
+    }
+
+    @Test
+    @Order(10)
+    public void whenSelectListIsInvoked_ThenResultsAreStreamed() {
+        clientService.getBaseMapper()
+            .selectList(Wrappers.emptyWrapper(), resultContext ->
+                assertNotNull(resultContext.getResultObject()));
+    }
+
+    @Test
+    @Order(11)
+    public void givenExistingClient_WhenUsingRemovedByIdWithDeleted_ThenClientIsLogicallyDeleted() {
+        Map<String, String> map = Map.of("first_name", "Harry");
+
+        QueryWrapper<Client> clientQueryWrapper = new QueryWrapper<>();
+        clientQueryWrapper.allEq(map);
+        Client harry = clientService.getBaseMapper()
+            .selectOne(clientQueryWrapper);
+
+        clientService.removeById(harry);
+
+        assertNull(clientService.getById(harry.getId()));
+    }
+
+    @Test
+    @Order(12)
+    public void whenNextIdIsInvoked_ThenCustomIdGeneratorIsUsed() {
+        Client harry = new Client();
+        harry.setId(timestampIdGenerator.nextId(harry));
+        harry.setFirstName("Harry");
+        clientService.save(harry);
+
+        assertTrue(System.nanoTime() > harry.getId());
```

Comment:
I don't understand this assertion?

Reply:
changed it to
`assertTrue(timestampIdGenerator.nextId(harry) > harry.getId(),
            "Since we've used the timestampIdGenerator, the NextId value is greater than the previous Id");`